src/com/cdd/bao/template/Vocabulary.java

```diff
@@ -714,4 +738,17 @@ private void addEquivalence(String uri1, String uri2)
 		}
 		else equivalence.put(uri1, new String[]{uri2});
 	}
+
+	// if input URI is remapped, then return last URI in remapping sequence; otherwise, simply return the input URI
+	// NOTE: assume that there is no cycle in remappings by the time this method is invoked
+	private String remapIfAny(String lastUri)
+	{
+		String nextUri = remappings.get(lastUri);
+		while (nextUri != null)
+		{
+			lastUri = nextUri;
+			nextUri = remappings.get(nextUri);
+		}
```

Comment:
```
private String remapIfAny(String uri)
{
  while (remappings.hasKey(uri)) uri = remappings.get(uri);
  return uri;
}
```

Reply:
Implementing these changes shortly.

Reply:
Also don't use the acronym-to-mixedcaps naming style (there's a reason that Microsoft likes to encourage that, but we don't use any of the features that drive that). So they should be lastURI and nextURI.